src/repository/models.py

```diff
@@ -623,6 +632,52 @@ def local_url(self):
 
         return url
 
+    def create_article(self, journal, workflow_stage, journal_license, journal_section, force=False):
+        """
+        Creates an article in a given journal and workflow stage.
+        """
+        if not self.article or force:
+            # create base article
+            article = submission_models.Article.objects.create(
+                journal=journal,
+                owner=self.owner,
+                title=self.title,
+                abstract=self.abstract,
+                license=journal_license,
+                section=journal_section,
+                date_submitted=timezone.now(),
+                comments_editor='Submitted from {}'.format(self.repository.name),
+                stage=workflow_stage,
+            )
+
+            # copy authors to submission
+            for preprint_author in self.preprintauthor_set.all():
+                submission_models.ArticleAuthorOrder.objects.get_or_create(
+                    article=article,
+                    author=preprint_author.account,
+                    defaults={
+                        'order': preprint_author.order
+                    }
+                )
+
+            # snapshot authors
+            article.snapshot_authors()
+
+            # copy preprints latest file and add it as a MS file to the article
+            file = files.copy_preprint_file_to_article(
+                self,
+                article,
+            )
+            article.manuscript_files.add(file)
+
+            # save and return the article
+            self.article = article
```

Comment:
I'm a bit confused about what the `force` argument does here, presumably it leaves any previously linked article orphaned from the preprint, is this a behaviour we want? Wouldn't it be best to first ask the editor to delete the article or unlink it from the preprint explicitely?

Reply:
Yes it will unlink the other article so there and replace it with the new one. If they want to delete the article thats another story (no other area of Janeway gives anyone the ability to delete an article except via admin), I think, but isn't this unlink pretty explicit given the help text that runs along with the field?

Reply:
I think, as you comment below, making it explicitly clear what happens to the old article in the help text will clear this up.